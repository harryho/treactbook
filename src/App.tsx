import * as React from 'react';
import './App.css';
import { RoundedButton } from './components/RoundedButton';

import { connect } from 'react-redux';
import { AppState } from './store';

import { SystemState, ChatState } from './store/types';
import { updateSession, sendMessage } from './store/actions';

import ChatHistory from './components/ChatHistory';
import ChatInput from './components/ChatInput';

import { thunkSendMessage } from './services/thunks';

const helloFunc = () => {
  alert('Hello');
};


interface AppProps {
  sendMessage: typeof sendMessage;
  updateSession: typeof updateSession;
  chat: ChatState;
  system: SystemState;
  thunkSendMessage: any;
}

export type UpdateMessageParam = React.SyntheticEvent<{ value: string }>;

class App extends React.Component<AppProps> {
  state = {
    message: ''
  };

  componentDidMount() {
    this.props.updateSession({
      loggedIn: true,
      session: 'my_session',
      userName: 'myName'
    });
    this.props.sendMessage({
      user: 'Chat Bot',
      message: 'This is a very basic chat application written in typescript using react and redux. Feel free to explore the source code.',
      timestamp: new Date().getTime()
    });
    this.props.thunkSendMessage('This message was sent by a thunk!');
  }

  updateMessage = (event: UpdateMessageParam) => {
    this.setState({ message: event.currentTarget.value });
  };

  sendMessage = (message: string) => {
    this.props.sendMessage({
      user: this.props.system.userName,
      message: message,
      timestamp: new Date().getTime()
    });
    this.setState({ message: '' });
  };

  render() {
    return (
      <div className="parent">
        <header className="App-header">
          <p>
            Edit <code>src/App.tsx</code> and save to reload.
          </p>

          <a className="App-link" href="https:reactjs.org" target="_blank" rel="noopener noreferrer">
            Learn React
          </a>
          <RoundedButton color="maroon" onClick={helloFunc}>
            New Button
          </RoundedButton>
          <ChatHistory messages={this.props.chat ? this.props.chat.messages : []} />
          <ChatInput
            userName={this.props.system ? this.props.system.userName : ''}
            message={this.state.message}
            updateMessage={this.updateMessage}
            sendMessage={this.sendMessage}
          />
        </header>
      </div>
    );
  }
}

const mapStateToProps = (state: AppState) => ({
  system: state.system,
  chat: state.chat
});

export default connect(mapStateToProps, {
  sendMessage,
  updateSession,
  thunkSendMessage
})(App);

