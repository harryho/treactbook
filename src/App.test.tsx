import React from 'react';
// import { render } from '@testing-library/react';
import {create, act} from 'react-test-renderer';
import App from './App';
import { Provider } from "react-redux";
// import configureStore from 'redux-mock-store';
import configureStore from "./store";

const mockStore = configureStore();

let store;
beforeEach(() => {
  store = mockStore
});

test('renders learn react link', () => {
  let app;
  act(() => {
    app = create(
    <Provider store={store}>
      <App />
    </Provider>
    );
  });

  expect(app.toJSON()).toMatchSnapshot();

});
