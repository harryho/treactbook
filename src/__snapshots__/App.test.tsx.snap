// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="parent"
>
  <header
    className="App-header"
  >
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https:reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "maroon",
          "border": "none",
          "borderRadius": 5,
          "color": "white",
          "display": "inline-block",
          "fontSize": "16px",
          "margin": "4px 2px",
          "padding": 20,
          "textAlign": "center",
          "textDecoration": "none",
        }
      }
    >
      New Button
    </button>
    <div
      className="chat-history"
    >
      <div
        className="message-item"
      >
        <h3>
          From: 
          Chat Bot
        </h3>
        <p>
          This is a very basic chat application written in typescript using react and redux. Feel free to explore the source code.
        </p>
      </div>
    </div>
    <div
      className="chat-interface"
    >
      <h3>
        User: 
        myName
         
      </h3>
      <input
        className="chat-input"
        onChange={[Function]}
        onKeyPress={[Function]}
        placeholder="Type a message..."
        value=""
      />
      <button
        onClick={[Function]}
      >
        Send
      </button>
    </div>
  </header>
</div>
`;
